source/administration/mmctl-cli-tool.rst

```diff
@@ -0,0 +1,2346 @@
+mmctl Command Line Tool (Beta)
+==============================
+
+The mmctl tool is a remote CLI tool for Mattermost which is installed locally and uses the Mattermost API. Authentication
+is done with either login credentials or an authentication token.
+
+Being installed locally allows a System Admin to run CLI commands even in instances where there is no access to the
+server (e.g., via SSH). This tool is currently in beta and can be used alongside the Mattermost CLI tool.
+In the future, the Mattermost CLI tool will be deprecated.
+
+This feature was developed to a large extent by community contributions and we'd like to extend our gratitude to the contributors who have
+worked on this project. We are currently accepting PRs for Help Wanted issues
+in the `mattermost-server <https://github.com/mattermost/mattermost-server/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+mmctl>`__ repo. You can learn more about
+the unit test coverage campaign for mmctl in the `Unit testing mmctl commands <https://mattermost.com/blog/unit-testing-mmctl-commands/>`__ blog post.
+
+
+
+**Notes**
+
+-  Parameters in CLI commands are order-specific.
+-  If special characters (``!``, ``|``, ``(``, ``)``, ``\``, ``'``, and ``"``) are used, the entire argument needs to be surrounded by single quotes (e.g. ``-password 'mypassword!'``, or the individual characters need to be escaped out (e.g. ``password mypassword\!``).
+- Team name and channel name refer to the handles, not the display names. So in the URL ``https://community.mattermost.com/core/channels/town-square`` team name would be ``core`` and channel name would be ``town-square``.
+
+
+**Options**
+
+.. code-block:: sh
+
+       --format string    the format of the command output [plain, json] (default "plain")
+       -h, --help         help for mmctl
+
+**Commands**
+   - `mmctl auth`_ - Authentication Management
+   - `mmctl channel`_ - Channel Management
+   - `mmctl command`_ - Command Management
+   - `mmctl config`_ - Configuration Management
+   - `mmctl group`_ - Group Management
+   - `mmctl ldap`_ - LDAP Management
+   - `mmctl license`_ - License Management
+   - `mmctl logs`_ - Log Management
+   - `mmctl permissions`_ - Permissions Management
+   - `mmctl plugin`_ - Plugin Management
+   - `mmctl post`_ - Post Management
+   - `mmctl roles`_ - Roles Management
+   - `mmctl team`_ - Team Management
+   - `mmctl user`_ - User Management
+   - `mmctl version`_ - Version Management
+   - `mmctl webhook`_ - Webhook Management
+
+
+Installing mmctl
+----------------
+
+To install the project in your `$GOPATH` run the following command as admin user:
```

Comment:
We can include here the use of `brew install mmctl` command too. What do you think @mgdelacroix ?

Reply:
This is a good suggestion. Would you suggest breaking this into sub-sections for different OSs or including it as an alternative in a single sentence?

Reply:
That would be great. We could do it by OS or we can have two sections one for building the tool and the other one to install from package manager

Reply:
I agree that separating the build and the install process would be good. I think that for the different install processes we can create a "Methods" section, and then describe ways to install it "via go get", "via brew", "via release package", etc. Chances are that we end up having more package managers to add here, and as some OSs share install methods (release and go get are for all of them, brew is for OSX and Linux...) I think having them under "Methods" suits it better

Reply:
Great, thank you! Would you be open to sharing the methods and I can add them to the doc and do the restructuring?

Reply:
Sure, right now we have:

### Via brew

Valid for Linux and Mac, with brew installed, running `brew install mmctl` will install the tool.

### Via go get

Valid for Linux, Mac and Windows, with the go environment configured, running `go get -u github.com/mattermost/mmctl` would install the tool.

### Via release package

Valid for Linux, Mac and Windows, user needs to go to the [mmctl releases page](https://github.com/mattermost/mmctl/releases) and download the appropriate release for their OS, which will contain the binary of the tool.

Reply:
I've updated the doc with this feedback.